cluster-creation-cli/README.md

```diff
@@ -0,0 +1,353 @@
+
+- `--provider`: (required) provider name. For example: capa, capvcd, capz, eks.
+  - TODO: How do we restrict and validate the provider list per installation?
+- `--config`: Path to a config file (optional). Can be used multiple times. See "Configuration layering" below. As an alternative, or in addition, configuration can be passed via standard input.
+- `--release`: the version number of the release to use. The release specifies the version of components to use in the cluster. If not specified, the latest release is used.
+- `--namespace` / `-n`: the namespace in which to create the cluster. This is expected to be an organization namespace, like `org-myorg`.
+- `--dry-run`: only validate the configuration and optionally print the resulting manifest.
+- `--output`: This flag can be used to specify the format of the output when `--dry-run` is set. See the "Dry run output" section below for details.
+  - `yaml` for a Kubernetes manifest that includes an App resource and one or several ConfigMap resources.
+  - `config-yaml` for merged configuration YAML.
+  - `command` for a self-contained representation of the command just executed, as a one-liner.
+- `--set`: Override a configuration value. This flag can be used multiple times. The format is `path.to.property=value`.
+  - Note: `--set` can only be used with scalar values (string, number, boolean, integer, etc.). It cannot be used to override an array or object.
+
+### Examples
+
+Ex. 1: Simple cluster creation based on a file
+
+    kubectl gs create cluster \
+      --provider capa \
+      --namespace org-test \
+      --config myconfig.yaml
+
+Ex. 2: Simple cluster creation based on standard input
+
+    kubectl gs create cluster \
+      --namespace org-test \
+      --provider capa <<EOF
+    global:
+      metadata:
+        servicePriority: lowest
+        name: test01
+        description: Just a test cluster
+    EOF
+
+While ex. 1 appears simpler and allows for easier re-use of a file-based configuration for a single user, ex. 2 has the benefit of being completely self-contained. It's a pure one-liner without the need for a separate file. This way it's easy to share the command, or copy the command from a website or web UI and paste it into a terminal.
+
+Ex. 3: Overwrite a configuration value
+
+    kubectl gs create cluster \
+      --provider capa \
+      --namespace org-test \
+      --config myconfig.yaml \
+      --set global.metadata.name=test02
+
+Ex. 4: Dry run
+
+    kubectl gs create cluster \
+      --provider capa \
+      --namespace org-test \
+      --config myconfig.yaml \
+      --dry-run
+
+Ex. 5: Write the result to a file
+
+    kubectl gs create cluster \
+      --provider capa \
+      --namespace org-test \
+      --config myconfig.yaml \
+      --dry-run \
+      --output yaml > manifest.yaml
+
+## Dry run output
+
+Combining `--dry-run` and `--output`, users can decide to print a representation of the desired cluster instead of creating it immediately.
```

Comment:
For existing users, it's confusing that the command defaults to creating a cluster instead of printing a manifest. In the best sense of a Unix tool, it should do only one thing well, and rely on other tools (namely `| kubectl apply -f -`) to do out-of-scope actions. With GitOps being the overall goal for most users, it's quite surprising and unsafe to have the new command create a workload cluster immediately. CLIs should not use `--something` to change to a different mode – here, I'd expect `--output yaml` to print a manifest, which clearly is a different mode from applying a cluster manifest.

Also, we have lots of existing users of `kubectl gs template cluster` who already rely on certain flags. We'd need to migrate them to the new command and deprecate the command line arguments, or maintain the old command forever. Therefore, old and new command should be compatible. Same for internal users.

Reply:
> For existing users, it's confusing that the command defaults to creating a cluster instead of printing a manifest.

I think this is the case when you think of `template cluster` as a reference.

Since this is a new command named `create` cluster, I think it's fair to assume that users understand the intention is to actually create a cluster.